### PR TITLE
Add developer shell for Rust programming language

### PR DESCRIPTION
**Pull request by yvan-sraka**
You can try it with `nix develop "github:yvan-sraka/devx#rust"` 😉!

### CODE REVIEW

File: fix-devshells.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#
+# … you will get an error!
+#
+# Because Nix is looking for something like:
+#
+#     devShells.rust-1."66"."1" = mkShell { … };
+#
+# So below here is the right amount of Nix magic needed to turn a given
+# expression (like the first example) into the wanted expression (like the
+# second example) to work with `nix develop`!
```
Comment (angerman) [inline fix-devshells.nix:20]:
```suggestion
# The following code will therefore transform an attribute set of the form
#    { "rust-1.66.1" = expr; }
# into
#    { "rust-1.66.1" = expr;
#      rust-1 = { "66" = { "1" = expr; }; };
#    }
```

**Review by angerman — CHANGES_REQUESTED** [review]
I don't think this PR should contain the

- .editorconfig
- .envrc
- .gitignore

changes. They are orthogonal to the addition of rust support.

**Comment by yvan-sraka** [comment]
> I don't think this PR should contain the
> 
> * .editorconfig
> * .envrc
> * .gitignore
> 
> changes. They are orthogonal to the addition of rust support.

Do I open one PR by extra config file, are rather make a bulk one with all these little tweaks?

**Comment by angerman** [comment]
> > I don't think this PR should contain the
> 
> > 
> 
> > * .editorconfig
> 
> > * .envrc
> 
> > * .gitignore
> 
> > 
> 
> > changes. They are orthogonal to the addition of rust support.
> 
> 
> 
> Do I open one PR by extra config file, are rather make a bulk one with all these little tweaks?

Let's have a different PR with those and discuss the merits of adding them or not there.

**Comment by yvan-sraka** [comment]
> Let's have a different PR with those and discuss the merits of adding them or not there.

https://github.com/input-output-hk/devx/pull/27 :)

**Comment by angerman** [comment]
@yvan-sraka please rebase this, clean it up, and let's get this merged.

File: rust.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#
+# You can specify a channel by replacing `#rust` by `#rust-beta` or
+# `#rust-nightly` (by default the shell will use the latest stable), or
+# directly specify a Rust version, e.g., `#rust-1.66.1`.
+
+{ pkgs ? import <nixpkgs> { overlays = [ (import <rust-overlay>) ]; } }:
+with builtins;
+let
+  rust-channels = [ "stable" "beta" "nightly" ];
+  rust-versions = [
```
Comment (yvan-sraka) [inline rust.nix:18]: The Rust versions list should be updated!

**Pull request closed without merging**